addons/controllers/csi/vspherecsiconfig_utils.go

```diff
@@ -385,11 +385,9 @@ func (r *VSphereCSIConfigReconciler) isStretchedSupervisorCluster(ctx context.Co
 	}
 
 	if len(azList.Items) == 1 {
-		// by default every cluster contains legacy availability zone,
-		// if only the legacy AZ is present, return false
-		if azList.Items[0].Name == LegacyZoneName {
-			return false, nil
-		}
+		// by default every non-stretched cluster contains a single zone,
+		// if there is only one AZ we assume it is non-stretched cluster
+		return false, nil
```

Comment:
nit: Would it be a good idea to combine L387 with L383 by using just one IF statement ? That helps prevent `cyclomatic complexity` from exceeding the limit (I know it is not at the moment, but recommend we do our best to simplify the function).

Reply:
makes sense. Handled it ...